core/domain/exp_services_test.py

```diff
@@ -1222,6 +1223,51 @@ def test_save_original_and_compressed_versions_of_image(self):
         self.assertEqual(
             fs.isfile('image/%s' % self.MICRO_IMAGE_FILENAME), True)
 
+    def test_save_original_and_compressed_versions_of_image_with_dev_mode_off(
+            self):
+        dev_mode_swap = self.swap(constants, 'DEV_MODE', False)
+        max_resize_dimension_px_swap = self.swap(
+            gae_image_services, 'MAX_RESIZE_DIMENSION_PX', 20)
+        with open(os.path.join(feconf.TESTS_DATA_DIR, 'img.png')) as f:
+            original_image_content = f.read()
+
+        with dev_mode_swap, max_resize_dimension_px_swap:
+            fs = fs_domain.AbstractFileSystem(
+                fs_domain.GcsFileSystem(
+                    'exploration/%s' % self.EXPLORATION_ID))
+
+            self.assertEqual(fs.isfile('image/%s' % self.FILENAME), False)
+            self.assertEqual(
+                fs.isfile('image/%s' % self.COMPRESSED_IMAGE_FILENAME), False)
+            self.assertEqual(
+                fs.isfile('image/%s' % self.MICRO_IMAGE_FILENAME), False)
+
+            exp_services.save_original_and_compressed_versions_of_image(
+                self.USER, self.FILENAME, self.EXPLORATION_ID,
+                original_image_content)
+
+            # The scaling factor changes if the dimensions of the image is
+            # greater than MAX_RESIZE_DIMENSION_PX.
+            width, height = gae_image_services.get_image_dimensions(
```

Comment:
Tests shouldn't have logic. Retrieve each of the three images from the datastore and compare against hardcoded widths/heights for each. That way, your test is more robust against future changes in implementation. The hardcoded widths/heights will speak for themselves (an observer should be able to clearly see that they're different).

Reply:
Done